server/model/remandChangeModel.ts

```diff
@@ -18,4 +19,31 @@ export default class RemandChangeModel {
   public remandRelatedValidationSummary() {
     return remandRelatedValidationSummary(this.calculatedUnusedDeductions?.validationMessages)
   }
+
```

Comment:
ideally we try and use more functional programming techniques in modern typescript, e.g. moving away from using traditional for loops and mutable variables.. something like this. 

```
public isModified(): boolean {
  if (!this.dbAdjustment) {
    return false;
  }

  const sessionCharges = this.adjustment.remand.chargeId.sort((a, b) => a - b);
  const dbCharges = this.dbAdjustment.remand.chargeId.sort((a, b) => a - b);

  const chargeIdModified = !sessionCharges.every((chargeId, index) => chargeId === dbCharges[index]);

  const dateModified = this.adjustment.toDate !== this.dbAdjustment.toDate || 
                       this.adjustment.fromDate !== this.dbAdjustment.fromDate;

  return chargeIdModified || dateModified;
}

```

Reply:
I've adjusted this as you've suggested